lighthouse-core/lib/traces/tracing-processor.js

```diff
@@ -246,6 +235,28 @@ class TraceProcessor {
   }
 
   /**
+   * Provides the top level events on the main thread.
+   * @param {!traceviewer.Model} model
+   * @param {{traceEvents: !Array<!Object>}} trace
+   * @param {number=} startTime Optional start time (in ms) of range of interest. Defaults to trace start.
+   * @param {number=} endTime Optional end time (in ms) of range of interest. Defaults to trace end.
+   * @return {!Array<{start: number, end: number}>}
```

Comment:
`slice.duration` is used in `getMainThreadTopLevelEventDurations`, not sure if there's anything else needed on these objects